File: kinotic-js/workspace/packages/management-api/src/api/model/MicroserviceDeployment.ts
import type { Identifiable } from '@kinotic-ai/core'
import type { DeploymentStatus } from '@/api/model/DeploymentStatus'

/**
 * The standing deployment of one microservice artifact of a Project: the VM running it, the
 * machine identity that VM connects as, the commit it was last ensured for, and its status.
 * One row per microservice a deployment has ensured; a row outlives the artifact until the
 * deployment is removed.
 */
export class MicroserviceDeployment implements Identifiable<string> {

    /**
     * Unique id of the deployment.
     */
    public id: string | null = null

    public organizationId!: string

    public applicationId!: string

    /**
     * The id of the project the microservice belongs to.
     */
    public projectId!: string

    /**
     * The microservice's identity: the MicroserviceArtifact name it was deployed from. Unique
     * among the project's microservice deployments.
     */
    public name!: string

    /**
     * The id of the workload running the microservice, or null when the deployment could not
     * create one.
     */
    public workloadId: string | null = null

    /**
     * The id of the machine identity the microservice's workload authenticates as. Its secret
     * is issued once, with the workload it belongs to.
     */
    public machineIdentityId: string | null = null

    /**
     * The module the workload was started with, relative to the checkout root: the artifact's
     * directory joined with its entry. A commit that moves the entry point replaces the
     * workload.
     */
    public entryPoint: string | null = null

    /**
     * Sha of the commit the deployment was last ensured for.
     */
    public commitSha: string | null = null

    public status!: DeploymentStatus

    public created: number | null = null

    public updated: number | null = null

}

File: kinotic-management-api/src/main/java/org/kinotic/management/api/model/MicroserviceDeployment.java
package org.kinotic.management.api.model;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;
import org.kinotic.domain.api.model.DeploymentStatus;
import org.kinotic.domain.api.model.OrganizationScoped;

import java.util.Date;

/**
 * The standing deployment of one microservice artifact of a {@link Project}: the VM running
 * it, the machine identity that VM connects as, the commit it was last ensured for, and its
 * status. One row per microservice a deployment has ensured; a row outlives the artifact until
 * the deployment is removed.
 */
@Getter
@Setter
@Accessors(chain = true)
@NoArgsConstructor
public class MicroserviceDeployment implements OrganizationScoped<String> {

    /**
     * Unique id of the deployment.
     */
    private String id;

    private String organizationId;

    private String applicationId;

    /**
     * The id of the project the microservice belongs to.
     */
    private String projectId;

    /**
     * The microservice's identity: the {@link MicroserviceArtifact#name()} it was deployed
     * from. Unique among the project's microservice deployments.
     */
    private String name;

    /**
     * The id of the workload running the microservice, or {@code null} when the deployment
     * could not create one.
     */
    private String workloadId;

    /**
     * The id of the machine identity the microservice's workload authenticates as. Its secret
     * is issued once, with the workload it belongs to.
     */
    private String machineIdentityId;

    /**
     * The module the workload was started with, relative to the checkout root: the artifact's
     * directory joined with its entry. A commit that moves the entry point replaces the
     * workload.
     */
    private String entryPoint;

    /**
     * Sha of the commit the deployment was last ensured for.
     */
    private String commitSha;

    private DeploymentStatus status;

    private Date created;

    private Date updated;

}
